rllib/utils/torch_utils.py

```diff
@@ -172,6 +172,21 @@ def mapping(item):
     return tree.map_structure(mapping, x)
 
 
+@PublicAPI
+def copy_and_move_to_device(x: TensorStructType, device: Optional[str] = None):
```

Comment:
I tried using convert_to_torch_tensor when reloading optimizer state dicts, but it was doing something funny that was causing some of my types to get improperly cast, which caused a precision error down the line. Instead I created this function, which probably also deserves its own test.

Reply:
Is the funny thing something we should fix? Not saying that we should, just asking for your optinion.